libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -65,13 +66,12 @@ public static EntitlementChecker checker() {
     public static void initialize(Instrumentation inst) throws Exception {
         manager = initChecker();
 
-        Map<MethodKey, CheckMethod> checkMethods = INSTRUMENTER_FACTORY.lookupMethodsToInstrument(
-            "org.elasticsearch.entitlement.bridge.EntitlementChecker"
-        );
+        Map<MethodKey, CheckMethod> checkMethods = INSTRUMENTER_FACTORY.lookupMethods(EntitlementChecker.class);
 
         var classesToTransform = checkMethods.keySet().stream().map(MethodKey::className).collect(Collectors.toSet());
 
-        inst.addTransformer(new Transformer(INSTRUMENTER_FACTORY.newInstrumenter(checkMethods), classesToTransform), true);
+        Instrumenter instrumenter = INSTRUMENTER_FACTORY.newInstrumenter(EntitlementChecker.class, checkMethods);
```

Comment:
Is there no need to reference Java23EntitlementChecker anymore?

Reply:
Oof, looks like we already lost this. I'll fix in a followup, but I think we can still use the Class object here, we will want to do Class.forName anyways

Reply:
OK, but still, I do not see a way to load `Java23EntitlementChecker` in the new code. And I wonder if referencing a class directly (instead by name) could be an issue here (for JDK 22, for example).
I agree, using the Class object is nicer, but does it work?

Reply:
In a followup I will change this back to using a helper method. In the case of Java 23, it will use Class.forName to get the Class object, from a string. In the default case it will use the `.class` as it does here. In either case, we will pass in the `Class` object. That's what InstrumentationService was doing internally anyways.